ClangTidy.cmake

```diff
@@ -2,26 +2,30 @@
 
 # Setup clang-tidy target (enabled if toolchain is clang)
 message(STATUS "Setup target: clang-tidy")
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+    find_program(CLANG_TIDY_EXE NAMES "clang-tidy" PATHS /usr/local/opt/llvm/bin)
+    find_program(RUN_CLANG_TIDY_EXE NAMES "run-clang-tidy.py" PATHS /usr/local/Cellar/llvm/7.0.1/share/clang)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     find_program(CLANG_TIDY_EXE NAMES "clang-tidy-7" PATHS /usr/bin)
     find_program(RUN_CLANG_TIDY_EXE NAMES "run-clang-tidy.py" PATHS /usr/lib/llvm-7/share/clang)
+else()
+    message(STATUS "Skipping setup of clang-tidy. Toolchain is not clang.")
+endif()
 
-    if (NOT CLANG_TIDY_EXE)
-        message(WARNING "clang-tidy not found!")
-    endif()
+if (NOT CLANG_TIDY_EXE)
+    message(WARNING "clang-tidy not found!")
```

Comment:
You changed logic. Now if I'll build with gcc I'll two warning. This is not correct.

Reply:
Oops, my fault. Anyway, reverted back.